### PR TITLE
[IE TESTS] Add OneHot SLT

### PR DESCRIPTION
**Pull request by IvanNovoselov**

### CODE REVIEW

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/one_hot.cpp [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        testing::ValuesIn(argOnValue_T),
+        testing::ValuesIn(argOffValue_T),
+        testing::ValuesIn(argAxis_T),
+        testing::ValuesIn(netPrecisions),
+        testing::ValuesIn(inputShapes_T),
+        testing::Values(CommonTestUtils::DEVICE_CPU)
+);
+
+INSTANTIATE_TEST_CASE_P(
+        OneHotArgType,
```
Comment (iefode) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/one_hot.cpp:95]: Are you sure, that it should be out of pre-commit scope?

File: inference-engine/tests/ngraph_helpers/ngraph_functions/src/one_hot.cpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+                                          const element::Type& depth_type,
+                                          const int64_t& depth_val,
+                                          const element::Type& set_type,
+                                          const float& on_val,
+                                          const float& off_val,
+                                          const int64_t& axis) {
+    auto depth_const = std::make_shared<ngraph::op::Constant>(depth_type, ngraph::Shape{ }, depth_val);
+    auto on_value_const = std::make_shared<ngraph::op::Constant>(set_type, ngraph::Shape{ }, on_val);
+    auto off_value_const = std::make_shared<ngraph::op::Constant>(set_type, ngraph::Shape{ }, off_val);
+    return std::make_shared<ngraph::opset3::OneHot>(indices, depth_const, on_value_const, off_value_const, axis);
```
Comment (iefode) [inline inference-engine/tests/ngraph_helpers/ngraph_functions/src/one_hot.cpp:22]: `opset5`

File: inference-engine/tests/ngraph_helpers/ngraph_functions/src/utils/ngraph_helpers.cpp [hunk 3]
```diff
@@ -388,6 +388,43 @@ std::vector<std::uint8_t> convertOutputPrecision(const std::vector<std::uint8_t>
                                                                                                         element::Type(toPrecision).get_type_name());
             }
         }
+        case element::Type_t::u32: {
```
Comment (iefode) [inline inference-engine/tests/ngraph_helpers/ngraph_functions/src/utils/ngraph_helpers.cpp:391]: Are you sure, that these changes are required?

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/one_hot.cpp (hunk 1 above)
Comment (IvanNovoselov) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/one_hot.cpp:95]: The plugin owner suggested that only i32 and f32 are worth checking in pre-commit, so I moved other types out of the pre-commit scope.

File: inference-engine/tests/ngraph_helpers/ngraph_functions/src/one_hot.cpp (hunk 2 above)
Comment (IvanNovoselov) [reply inference-engine/tests/ngraph_helpers/ngraph_functions/src/one_hot.cpp:22]: Thanks! Will fix it.

File: inference-engine/tests/ngraph_helpers/ngraph_functions/src/utils/ngraph_helpers.cpp (hunk 3 above)
Comment (IvanNovoselov) [reply inference-engine/tests/ngraph_helpers/ngraph_functions/src/utils/ngraph_helpers.cpp:391]: No, they are not required. I will roll back to the original version.

Comment (IvanNovoselov) [reply inference-engine/tests/ngraph_helpers/ngraph_functions/src/utils/ngraph_helpers.cpp]: No, the changes are not required. I will roll back the ngraph_helpers.cpp to the original version.

**Review by iefode — APPROVED** [review]

**Comment by iefode** [comment]
@dmitry-gorokhov Will you take look the PR?

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/one_hot.cpp (hunk 1 above)
Comment (dmitry-gorokhov) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/one_hot.cpp:95]: Maybe we had some miscommunication. I would say it is better to have all in pre-commit until it doesn't increase tests time significantly.

**Review by dmitry-gorokhov — APPROVED** [review]
In general looks good to me. Please move tests in precommit scope and make CI green.

**Pull request merged**